src/sinks/pulsar.rs

```diff
@@ -246,18 +275,34 @@ impl PulsarSinkConfig {
             });
         }
 
+        let mut producer_options = pulsar::ProducerOptions {
+            compression: Some(match self.compression {
+                PulsarCompression::None => compression::Compression::None,
+                PulsarCompression::Lz4 => {
+                    compression::Compression::Lz4(compression::CompressionLz4::default())
+                }
+                PulsarCompression::Zlib => {
+                    compression::Compression::Zlib(compression::CompressionZlib::default())
+                }
+                PulsarCompression::Zstd => {
+                    compression::Compression::Zstd(compression::CompressionZstd::default())
+                }
+                PulsarCompression::Snappy => {
+                    compression::Compression::Snappy(compression::CompressionSnappy::default())
+                }
+            }),
```

Comment:
This could also be implemented as `impl From<PulsarCompression> for compression::Compression` to keep this function shorter, but since it's the only usage it doesn't matter all that much.

Reply:
I think you could also nest each `compression::Compression` instance inside of each `PulsarCompression` enum value as a field?